src/ingestors/foundation/index.ts

```diff
@@ -0,0 +1,133 @@
+import { MintContractOptions, MintIngestor, MintIngestorResources } from '../../lib/types/mint-ingestor';
+import { MintIngestionErrorName, MintIngestorError } from '../../lib/types/mint-ingestor-error';
+import { MintInstructionType, MintTemplate } from '../../lib/types/mint-template';
+import { MintTemplateBuilder } from '../../lib/builder/mint-template-builder';
+import { getFoundationMintPriceInWei } from './onchain-metadata';
+import { getFoundationMintByAddress } from './offchain-metadata';
+import { FOUNDATION_MINT_ABI } from './abi';
+
+const CONTRACT_ADDRESS = '0x62037b26ffF91929655AA3A060F327b47d1e2b3e';
+
+const getChainId = (chain: string) => {
+  switch (chain) {
+    case 'base':
+      return 8453;
+    case 'eth':
+    default:
+      return 1;
+  }
+};
+
+export class FoundationIngestor implements MintIngestor {
+  async supportsUrl(_resources: MintIngestorResources, url: string): Promise<boolean> {
+    return new URL(url).hostname === 'www.foundation.app' || new URL(url).hostname === 'foundation.app';
+  }
+
+  async supportsContract(resources: MintIngestorResources, contractOptions: MintContractOptions): Promise<boolean> {
+    if (contractOptions.chainId !== 8453) {
+      return false;
+    }
+    const contract = await getFoundationMintByAddress(resources, contractOptions);
+    if (!contract) {
+      return false;
+    }
+    return true;
+  }
+
+  async createMintForContract(
+    resources: MintIngestorResources,
+    contractOptions: MintContractOptions,
+  ): Promise<MintTemplate> {
+    const mintBuilder = new MintTemplateBuilder()
+      .setMintInstructionType(MintInstructionType.EVM_MINT)
+      .setPartnerName('foundation');
+
+    if (contractOptions.url) {
+      mintBuilder.setMarketingUrl(contractOptions.url);
+    }
+
+    const collection = await getFoundationMintByAddress(resources, contractOptions);
+
+    if (!collection) {
+      throw new MintIngestorError(MintIngestionErrorName.CouldNotResolveMint, 'Collection not found');
+    }
+
+    const contractAddress = collection.contractAddress;
+
+    const description = collection?.description;
+    const image = collection?.media.url;
+
+    mintBuilder.setName(collection.name).setDescription(description).setFeaturedImageUrl(image);
+    mintBuilder.setMintOutputContract({ chainId: 8453, address: contractAddress });
+    mintBuilder.addImage(collection.banner_image_url, 'banner');
+
+    if (!collection.creator) {
+      throw new Error('Error finding creator');
+    }
+
+    mintBuilder.setCreator({
+      name: collection.creator.name,
+      walletAddress: collection.creator.publicKey,
+    });
+
+    const totalPriceWei = await getFoundationMintPriceInWei(
```

Comment:
can we try-catch this and raise a semantic error if not available. e.g. attempting an ETH mint throws out contract errors